server/src/test/document/DbtTextDocument.spec.ts

```diff
@@ -157,6 +157,19 @@ describe('DbtTextDocument', () => {
     verify(mockModelCompiler.compile(anything())).once();
   });
 
+  it('didOpenTextDocument should start compilation if previous compile stuck (document contains jinjas and raw document text is the same as compiled)', async () => {
+    // arrange
+    when(mockJinjaParser.hasJinjas(TEXT)).thenReturn(false);
+    when(mockJinjaParser.findAllJinjaRanges(document.rawDocument)).thenReturn([Range.create(0, 0, 1, 1)]);
+
+    // act
+    await document.didOpenTextDocument(false);
```

Comment:
If we should doing `didChangeTextDocument` and `didSaveTextDocument` here?

Reply:
Made some refactor.